src/store/consts/utils.ts

```diff
@@ -235,27 +244,37 @@ export const createPlaceholderLiquidityPlot = (
   return isXtoY ? ticksData : ticksData.reverse()
 }
 
-export interface CoingeckoPriceData {
-  price: number
-  priceChange: number
-}
-export interface CoingeckoApiPriceData {
+export type CoinGeckoAPIData = CoinGeckoAPIPriceData[]
+
+export type CoinGeckoAPIPriceData = {
   id: string
   current_price: number
   price_change_percentage_24h: number
 }
 
-export const getCoingeckoTokenPrice = async (id: string): Promise<CoingeckoPriceData> => {
-  return await axios
-    .get<
-      CoingeckoApiPriceData[]
-    >(`https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&ids=${id}`)
-    .then(res => {
-      return {
-        price: res.data[0].current_price ?? 0,
-        priceChange: res.data[0].price_change_percentage_24h ?? 0
-      }
-    })
+const COINGECKO_QUERY_COOLDOWN = 20 * 60 * 1000
+
+export const getCoingeckoTokenPrice = async (id: string): Promise<number | undefined> => {
+  const cachedLastQueryTimestamp = localStorage.getItem('COINGECKO_LAST_QUERY_TIMESTAMP')
+  let lastQueryTimestamp = Date.now()
+  if (cachedLastQueryTimestamp) {
+    lastQueryTimestamp = Number(lastQueryTimestamp)
```

Comment:
I also think this function should be refactored in terms of minimizing access to local storage and removing potential race conditions.

Additionally, it looks like it doesn't work according to the assumption that there should be no price updates if there was an update in the last 20 minutes. ChatGPT can help here